Registry.php

```diff
@@ -170,12 +169,25 @@ public function getEntityManagerForClass($class)
 
     public function reset() : void
     {
-        if (! interface_exists(LazyLoadingInterface::class)) {
+        foreach ($this->getManagerNames() as $managerName => $serviceId) {
+            $this->resetOrClearManager($managerName, $serviceId);
+        }
+    }
+
+    private function resetOrClearManager(string $managerName, string $serviceId) : void
+    {
+        if (! $this->container->initialized($serviceId)) {
             return;
```

Comment:
The method is not in the interface so it needs an extra check before calling it.

Reply:
The type hint for the `$container` property in `ManagerRegistry` specifies this as a `Container`, not `ContainerInterface`. Similarly, the `resetService` method there doesn't check for this either. We should either fix the type hint and calls in `ManagerRegistry`, or restrict the constructor type hint in `Registry` to no longer accept any `ContainerInterface`.

Given the BC implications of changing the constructor signature in the bundle, I'd rather fix this in the bridge. I can create the PR if you agree.

Reply:
Ok thanks for double checking. No need for more on my side.